### ISSUE TITLE
[Bug]: WasteNet Southland not working after 1.46.0

### ISSUE THREAD

**Issue by etamtlosz** [mampfes/hacs_waste_collection_schedule#1872]
### I Have A Problem With:

A specific source

### What's Your Problem

The WasteNet Southland website and url has changed about a month ago. The issue created by this change was supposed to be fixed in 1.46.0, but unfortunately it is still not working.
Tested with my address and even with the example data, returning all sensors as unknown.

### Source (if relevant)

wastenet_org_nz

### Logs

```Shell
no relevant logs
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by 5ila5** [comment]
Address "156 Tay Street INVERCARGILL" works on my side. Is your calendar populated? Maybe the entry names changed and you need to adjust your sensor config.

**Comment by etamtlosz** [comment]
That's weird. "156 Tay Street INVERCARGILL" is not working for me. I have tried other waste providers from other locations, with addresses I have found in the issues section and it worked with those.

I am using the following code in configuration.yaml

```waste_collection_schedule:
  sources:
    - name: wastenet_org_nz
      args:
        address: "156 Tay Street INVERCARGILL"
```

I have multiple calendars showing up (because of testing multiple addresses), most of them as `unavailable`,  while one is `off`. 
![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/7436784/cbb24bf6-30cf-431e-9972-74ce87256596)

`Gore, Invercargill & Southland` is available and ticked in my HA calendar tab, but nothing regarding waste pickup is showing up under any date. Is there anything else needed in the config?

Tried adding the following code, I had used in the past, but it didn't help much.
```
      customize:
        - type: Rubbish
          alias: rubbish
          show: true
        - type: Recycling
          alias: recycling
          show:
            true
            # icon: ICON
  day_switch_time: "23:59"
```

**Comment by 5ila5** [comment]
For just the calendar to work, your first config is sufficient. `off` means working, but the collection is not now. I noticed an error in the previous implementation: the date is parsed in the wrong format (y/d/m instead of d/m/y). Fixed that in #1879

This will be part of release 1.47.0 or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
